arch/RISCV/RISCVDecode.gen.inc

```diff
@@ -9412,14 +9430,510 @@ static void decode(struct ast *tree, uint64_t binary_stream, RVContext *ctx) {
+      tree->ast_node_type = RISCV_VCLMULH_VX;
+      tree->ast_node.vclmulh_vx.vm = vm;
+      tree->ast_node.vclmulh_vx.vs2 = vs2;
+      tree->ast_node.vclmulh_vx.rs1 = rs1;
+      tree->ast_node.vclmulh_vx.vd = vd;
+      return;
+    }
+  }
+  //------------------------------------------------------------------------------------//
+
+  //----------------------------VSHA2MS_VV------------------------------//
+  if (SLICE_BITVEC(binary_stream, 0, 6) == 0x77 &&
+      SLICE_BITVEC(binary_stream, 12, 14) == 0x2 &&
+      SLICE_BITVEC(binary_stream, 25, 31) == 0x5B) {
+    uint64_t vd = SLICE_BITVEC(binary_stream, 7, 11);
+    uint64_t vs1 = SLICE_BITVEC(binary_stream, 15, 19);
+    uint64_t vs2 = SLICE_BITVEC(binary_stream, 20, 24);
+    if (((currentlyEnabled(RISCV_Ext_Zvknha, ctx)) && (get_sew(ctx) == 32)) ||
+        (((currentlyEnabled(RISCV_Ext_Zvknhb, ctx)) &&
+          ((get_sew(ctx) == 32) || (get_sew(ctx) == 64))) &&
+         (zvknhab_check_encdec(vs2, vs1, vd, ctx)))) {
+      tree->ast_node_type = RISCV_VSHA2MS_VV;
+      tree->ast_node.vsha2ms_vv.vs2 = vs2;
+      tree->ast_node.vsha2ms_vv.vs1 = vs1;
+      tree->ast_node.vsha2ms_vv.vd = vd;
+      return;
+    }
+  }
+  //------------------------------------------------------------------------------------//
+
+  //----------------------------ZVKSHA2TYPE------------------------------//
+  if (SLICE_BITVEC(binary_stream, 0, 6) == 0x77 &&
+      SLICE_BITVEC(binary_stream, 12, 14) == 0x2 &&
+      SLICE_BITVEC(binary_stream, 25, 25) == 0x1) {
+    uint64_t funct6 = 0xFFFFFFFFFFFFFFFF;
+    switch (SLICE_BITVEC(binary_stream, 26, 31)) {
+    case 0x2E:
+      funct6 = RISCV_ZVK_VSHA2CH;
+      break;
+    case 0x2F:
+      funct6 = RISCV_ZVK_VSHA2CL;
+      break;
+    }
+    if (funct6 != 0xFFFFFFFFFFFFFFFF) {
+      uint64_t vd = SLICE_BITVEC(binary_stream, 7, 11);
+      uint64_t vs1 = SLICE_BITVEC(binary_stream, 15, 19);
+      uint64_t vs2 = SLICE_BITVEC(binary_stream, 20, 24);
+      if (((currentlyEnabled(RISCV_Ext_Zvknha, ctx)) && (get_sew(ctx) == 32)) ||
+          (((currentlyEnabled(RISCV_Ext_Zvknhb, ctx)) &&
+            ((get_sew(ctx) == 32) || (get_sew(ctx) == 64))) &&
+           (zvknhab_check_encdec(vs2, vs1, vd, ctx)))) {
+        tree->ast_node_type = RISCV_ZVKSHA2TYPE;
+        tree->ast_node.zvksha2type.funct6 = funct6;
+        tree->ast_node.zvksha2type.vs2 = vs2;
+        tree->ast_node.zvksha2type.vs1 = vs1;
+        tree->ast_node.zvksha2type.vd = vd;
+        return;
+      }
+    }
+  }
+  //------------------------------------------------------------------------------------//
+
+  //----------------------------ZIMOP_MOP_R------------------------------//
+  if (SLICE_BITVEC(binary_stream, 0, 6) == 0x73 &&
+      SLICE_BITVEC(binary_stream, 12, 14) == 0x4 &&
+      SLICE_BITVEC(binary_stream, 22, 25) == 0x7 &&
+      SLICE_BITVEC(binary_stream, 28, 29) == 0x0 &&
+      SLICE_BITVEC(binary_stream, 31, 31) == 0x1) {
```

Comment:
I know we talked about this before, but just mention it again so we don't forget it.
These chains of `if (conditions...)` are pretty bad for runtime.
But as said before, for now we can leave it like this and merge it like this into experimental.